app/models/concerns/calculation.rb

```diff
@@ -8,4 +8,20 @@ def minutes_of_current_week
     end_date = Time.now.end_of_week.to_date
     entries.where(date: start_date..end_date).sum(:minutes)
   end
+
+  included do
+    def self.calculate_leaderboard
+      start_date = Time.now.beginning_of_week.to_date
+      end_date = Time.now.end_of_week.to_date
+
+      self.all.each do |record|
+        leaderboard_klass = "#{self.name}Leaderboard".constantize
+        leaderboard = leaderboard_klass.find_or_create_by(
```

Comment:
@schmierkov Considering that you're doing this (which seems a bit weird) why don't you just include the concern in `ProjectLeaderboard` and `UserLeaderboard`? 

It seems more natural to call `ProjectLeaderboard.calculate` and `UserLeaderboard.calculate`